client/lcd/swagger-ui/swagger.yaml

```diff
@@ -521,9 +556,629 @@ paths:
+  /stake/pool:
+    get:
+      summary: Get the current state of the staking pool
+      tags:
+      - ICS21
+      produces:
+      - application/json
+      responses:
+        200:
+          description: OK
+        500:
+          description: Internal Server Error
+  /stake/parameters:
+    get:
+      summary: Get the current staking parameter values
+      tags:
+      - ICS21
+      produces:
+      - application/json
+      responses:
+        200:
+          description: OK
+        500:
+          description: Internal Server Error
+  /slashing/signing_info/{validator}:
+    get:
+      summary: Get sign info of given validator
+      description: Get sign info of given validator
+      produces:
+      - application/json
+      tags:
+      - ICS23
+      parameters:
+      - type: string
+        description: Bech32 validator public key
+        name: validator
+        required: true
+        in: path
+      responses:
+        '200':
+          description: OK
+        500:
+          description: Internal Server Error
+  /slashing/unjail:
+    post:
+      summary: Unjail a jailed validator
+      description: Send transaction to unjail a jailed validator
+      consumes:
+      - application/json
+      produces:
+      - application/json
+      tags:
+      - ICS23
+      parameters:
+      - in: query
+        name: simulate
+        description: if true, ignore the gas field and perform a simulation of a transaction, but don't broadcast it
+        required: false
+        type: boolean
+      - in: query
+        name: generate_only
+        description: if true, build an unsigned transaction and write it back
+        required: false
+        type: boolean
+      - description: ''
+        name: UnjailBody
+        in: body
+        required: true
+        schema:
+          type: object
+          properties:
+            base_req:
+              "$ref": "#/definitions/BaseReq"
+            validator_addr:
+              "$ref": "#/definitions/ValidatorAddress"
+      responses:
+        200:
+          description: OK
+          schema:
+            $ref: "#/definitions/BroadcastTxCommitResult"
+        500:
+          description: Internal Server Error
+  /gov/proposals:
+    post:
+      summary: Submit a proposal
+      description: Send transaction to submit a proposal
+      consumes:
+      - application/json
+      produces:
+      - application/json
+      tags:
+      - ICS22
+      parameters:
+      - in: query
+        name: simulate
+        description: if true, ignore the gas field and perform a simulation of a transaction, but don't broadcast it
+        required: false
+        type: boolean
+      - in: query
+        name: generate_only
+        description: if true, build an unsigned transaction and write it back
+        required: false
+        type: boolean
+      - description: valid value of `"proposal_type"` can be `"Text"`, `"ParameterChange"`, `"SoftwareUpgrade"`
```

Comment:
Our properties are in snake_case. I think all values and properties should therefor be in snake_case. The types should therefor be `text`, `parameter_change` and `software_upgrade` to be consistent.

Reply:
But actually, we will encounter error if we use `text` as proposal_type. Do you mean we should change `Text` to `text` in handler implementation?

Reply:
yes, the API should be independent from the backend implementation. the API should be stable where i.e. the proposal type `SoftwareUpgrade` might be renamed internally to `ProposalSoftwareUpgrade` or something. Any client would still be able to send the same proposal type. what do you think?

Reply:
Agree here

Reply:
OK, I will change the code according to your suggestions. Thanks

Reply:
Do we need to implement this change in command line?